rusqlite_migration/src/lib.rs

```diff
@@ -467,6 +467,72 @@ impl<'m> Migrations<'m> {
         user_version(conn).map(|v| self.db_version_to_schema(v))
     }
 
+    /// Returns the number of migrations that would be applied by [`Migrations::to_latest`].
+    /// The number returned may be negative, if more migrations were applied than the current
+    /// version of the program knows about.
+    ///
+    /// For most common scenarios, you should be able to just call
+    /// [`Migrations::to_latest`], which already checks the schema version.
+    ///
+    /// # Examples
+    ///
+    /// ## Backup before applying migrations
+    ///
+    /// One common use case for this function is ta take a backup of the database before applying
+    /// migrations, if any migrations would run.
+    ///
+    /// ```
+    /// use rusqlite_migration::{Migrations, M};
+    ///
+    /// let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+    /// let mut migrations: Migrations = Migrations::new(vec![
+    ///     M::up("CREATE TABLE animals (name TEXT);"),
+    ///     M::up("CREATE TABLE food (name TEXT);"),
+    /// ]);
+    ///
+    /// if migrations.migrations_to_apply(&conn).unwrap() != 0 {
+    ///     // Backup the database
+    ///
+    ///     migrations.to_latest(&mut conn).unwrap()
+    /// }
+    /// ```
+    ///
+    ///
+    /// ## Negative numbers
+    ///
+    /// ```rust
+    /// use rusqlite_migration::{Error, Migrations, M, MigrationDefinitionError};
+    ///
+    /// let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+    ///
+    /// // Initial set of migrations in, say, version 1 of the program
+    /// let mut ms = vec![
+    ///     M::up("CREATE TABLE animals (name TEXT);"),
+    ///     M::up("CREATE TABLE food (name TEXT);"),
+    /// ];
+    /// let migrations_v1 = Migrations::new(ms.clone());
+    ///
+    /// migrations_v1.to_latest(&mut conn).unwrap();
+    /// assert_eq!(migrations_v1.migrations_to_apply(&conn), Ok(0));
+    ///
+    /// // More migrations are added in, say, version 2
+    /// ms.push(M::up("CREATE TABLE plants (name TEXT);"));
+    /// let migrations_v2 =  Migrations::new(ms);
+    ///
+    /// migrations_v2.to_latest(&mut conn).unwrap();
+    /// // From the perspective of the old version of the program, one migration would need to be
+    /// // reversed.
+    /// assert_eq!(migrations_v1.migrations_to_apply(&conn), Ok(-1));
+    /// // Note that in this situation, to_latest will return an error, which you can handle how
+    /// // you see fit (maybe restoring one of those backups or prompting the user)
+    /// assert_eq!(migrations_v1.to_latest(&mut conn), Err(Error::MigrationDefinition(
+    ///     MigrationDefinitionError::DatabaseTooFarAhead
+    /// )));
+    /// ```
+    pub fn migrations_to_apply(&self, conn: &Connection) -> Result<i32> {
```

Comment:
Not thrilled about that name, should probably change it before merging.

Reply:
`pending_migrations`?
`count_pending_migrations`?

Reply:
Thanks, I used the first name!